### ISSUE TITLE
CCM failed to get node address from cloud provider that matches IPv6

### ISSUE THREAD

**Issue by Preisschild** [hetznercloud/hcloud-cloud-controller-manager#647]
### TL;DR

When enabling dualstack in kubelet nodes won't be initialized by hcloud-ccm because it can't find the IPv6 address in the cloud provider

### Expected behavior

Nodes are able to be initialized when using dualstack

### Observed behavior

New nodes (with dualstack enabled) aren't able to initialize and hcloud-ccm logs this error 

```
node_controller.go:240] error syncing 'xxx-control-plane-cpx31-zzdwc': failed to get node modifiers from cloud provider: provided node ip for node "xxx-control-plane-cpx31-zzdwc" is not valid: failed to get node address from cloud provider that matches ip: 2a01:xxx:xxx:xxx::1, requeuing`
````

(The x shows the full IP, I just censored it)

I think the issue might be that Hetzner's API shows the full /64 network, but the node-ip is set to a specific host of that network (`::1`)

`2a01:xxx:xxx:xxx::/64` - thats what Hetzner shows in the Console)
 
`2a01:xxx:xxx:xxx::1` - thats what is logged by hcloud-ccm and is set in the node's `.status.addresses`

### Minimal working example

kubelet is configured with `--node-ip=135.xxx.xxx.57,2a01:xxx:xxx:xxx::1` and hcloud-ccm has the `--feature-gates=CloudDualStackNodeIPs=true` argument set.

### Additional information

Hcloud-ccm version `v1.19.0`

Kubernetes version `v1.30.1`

I also use Talos `1.7.2` as the underlying Operating System, and it automatically configures `--node-ip` for me.

**Comment by apricote** [comment]
Looking at the code we already handle the case of `::1` for IPv6. Did you set `HCLOUD_INSTANCES_ADDRESS_FAMILY=dualstack` on hccm? Only if that is set, do we return any IPv6 address.
